pages/directory/index.tsx

```diff
@@ -1,13 +1,64 @@
-import { Box, Container, List, ListItem, Typography } from '@mui/material';
+import { Box, Container, CircularProgress, List, ListItem, Typography } from '@mui/material';
+import axios from 'axios';
 import { NextPage } from 'next';
 import React from 'react';
 
 import createAxiosInstance from 'client/lib/axios';
 
 //TODO: Type users once sequelize-typescript is added
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
-const Directory: NextPage<{ users: any[]; error: string }> = props => {
-  const { users, error } = props;
+const Directory: NextPage<{ users: any[]; totalPages: number; error: string }> = props => {
+  const { users, totalPages, error } = props;
+  const lastCardRef = React.useRef(null);
+  const [userData, setUserData] = React.useState(users);
+  const [currentPage, setCurrentPage] = React.useState(1);
+  const [totalPageNum, setTotalPageNum] = React.useState(totalPages);
+  const [isLoading, setIsLoading] = React.useState(false);
+
+  async function fetchUsers(page: number) {
+    try {
+      const response = await axios.get(`/api/users?page=${page}`);
+      return {
+        page: response.data.page,
+        total: response.data.totalPages,
+        users: response.data.users,
+      };
+    } catch (err) {
+      console.error(err);
+    }
+  }
+
+  React.useEffect(() => {
+    const intersectionOptions = {
+      // root default = document's viewport
+      // in the future offset should be tied to the card height of the last row
+      rootMargin: '300px',
+      threshold: 0,
+    };
```

Comment:
Instead of using a magic number here, could we select a card and calculate its height? Something along the lines of:

```js
const height = getElementHeight(document.querySelector('card selector'));

function getElementHeight(el) {
  const rect = el.getBoundingClientRect();
  const { marginTop, marginBottom } = getComputedStyle(el);

  return rect.height + parseFloat(marginTop) + parseFloat(marginBottom);
}
```